controllers/apps/transformer_cluster_service.go

```diff
@@ -321,6 +321,9 @@ func createOrUpdateService(ctx graph.TransformContext, dag *graph.DAG, graphCli
 func resolveServiceDefaultFields(obj, objCopy *corev1.ServiceSpec) {
 	// TODO: how about the order changed?
 	for i, port := range objCopy.Ports {
+		if i == len(obj.Ports) {
```

Comment:
pls add a UT

Reply:
This looks like just a sanity check.

This loop tries to use the information in each entry of obj.Ports to complete the entries in objCopy.Ports. Therefore, it’s necessary to ensure that the indices in objCopy.Ports do not exceed the bounds of obj.Ports.

Reply:
So maybe it's unecessary to write UT for this sanity check.

Reply:
UT will be added in the future implementation~